lib/ansible/modules/network/iosxr/iosxr_interfaces.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for ios_interfaces
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+GENERATOR_VERSION = '1.0'
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                   'status': ['preview'],
+                   'supported_by': 'network'}
+
+NETWORK_OS = "iosxr"
+RESOURCE = "interfaces"
+COPYRIGHT = "Copyright 2019 Red Hat"
+
+DOCUMENTATION = """
+  module: iosxr_interfaces
+  version_added: 2.9
+  short_description: Manage interface attributes on Cisco IOS-XR network devices
+  description: This module manages the interface attributes on Cisco IOS-XR network devices.
+  author: Sumit Jaiswal (@justjais)
+  options:
+    config:
+      description: A dictionary of interface options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Full name of the interface to configure in C(type + path) format. e.g. C(GigabitEthernet0/0/0/0)
+          type: str
+          required: True
+        description:
+          description:
+          - Interface description.
+          type: str
+        enabled:
+          default: True
+          description:
+          - Administrative state of the interface.
+          - Set the value to C(True) to administratively enable the interface or C(False) to disable it.
+          type: bool
+        active:
+          default: active
+          description:
+          - Whether the interface is C(active) or C(preconfigured). Preconfiguration allows you to configure modular
+            services cards before they are inserted into the router. When the cards are inserted, they are instantly
+            configured. Active cards are the ones already inserted.
+          type: str
+          choices: ['active', 'preconfigure']
+        speed:
+          description:
+          - Configure the speed for an interface. Default is auto-negotiation when not configured.
+          type: str
+          choices: ['10', '100', '1000']
+        mtu:
+          description:
+          - Sets the MTU value for the interface. Range is between 64 and 65535. Applicable for Ethernet interfaces only.
```

Comment:
We should avoid mentioning ranges in docs. User can refer to vendor documentation for valid values.